### PR TITLE
Attach command: add Bazel filesystem support

### PR DESCRIPTION
**Pull request by DaveShuckerow**
Add a project root and filesystem options to the Attach command.  Test that it takes these parameters and passes them on to the FlutterDevice and HotRunner that it builds.

Cleanup: abstract out the common target options used by different commands into a single parent method.

### CODE REVIEW

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -122,6 +122,30 @@ abstract class FlutterCommand extends Command<Null> {
     _usesPubOption = true;
   }
 
+  /// Adds flags for using a specific filesystem root and scheme.
+  ///
+  /// Only usable in Dart 2 mode.
```
Comment (aam) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: This is redundant now as there is(or very shortly will be) only dart2 mode.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 2]
```diff
@@ -122,6 +122,30 @@ abstract class FlutterCommand extends Command<Null> {
...
+  ///
+  /// [hide] indicates whether or not to hide these options when the user asks
+  /// for help.
+  void usesFilesystemOptions({@required bool hide}) {
+    argParser..addOption('output-dill',
```
Comment (aam) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: formatting looks inconsistent - ..addOption, ..addMultiOption on new lines.

**Review by aam — APPROVED** [review]

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 2 above)
Comment (DaveShuckerow) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: Updated to be the same way it's done in attach.dart.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 1 above)
Comment (DaveShuckerow) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: I've pulled that comment out.

**Pull request merged**